### DISCUSSION TITLE
Java requirements - OpenApi

### DISCUSSION THREAD

**Discussion by thoroc** [aws/aws-pdk#759] [General]
Just a remark more than anything else. I like the idea to use NX and Projen together, but since I am not planning to use Smithy (which is Java base as far as I can tell), why the dependency on Java for OpenApi? That seems to be a bit heavy handed.

**Comment by agdimech** [comment]
Hi @thoroc - we use the [openapi generator](https://www.npmjs.com/package/@openapitools/openapi-generator-cli) to generate all of the types from an openapi spec for both smithy and openapi. For smithy, we actually convert the smithy spec into an openapi spec which is then passed into the openapi generator to generate all of the types. Given openapi generator is built using Java and that is the used in both variants, that is why it is needed unfortunately.

**Reply by thoroc** [reply]
I think that might hinder adoption where CI/CD pipelines are limiting what can be installed on the underlying container, if that makes sense. Never the less kudos on the project.
